packages/platforms/accton/arm64/as4224/src/modules-5_6/arm64-accton-as4224-psu.c

```diff
@@ -101,16 +110,40 @@ static int as4224_psu_read_value(u8 reg)
     return as4224_cpld_read(PSU_STATUS_I2C_ADDR, reg);
 }
 
+static int as4224_psu_write_value(u8 reg, u8 value)
+{
+    return as4224_cpld_write(PSU_STATUS_I2C_ADDR, reg, value);
+}
+
+static int as4224_psu_get_status_mask_reg(void)
+{
+    return (data->platform_id == AS5114_48X) ? 0x32 : 0x34;
+}
+
 static struct as4224_psu_data *as4224_psu_update_device(struct device *dev)
 {
     if (time_after(jiffies, data->last_updated + HZ + HZ / 2)
         || !data->valid) {
         int status;
+        int reg;
+        int reg_mask = 0x6;
 
         dev_dbg(dev, "Starting as4224 update\n");
         data->valid = 0;
 
         /* Read psu status */
+        reg = as4224_psu_get_status_mask_reg();
+        status = as4224_psu_read_value(reg);
+        if (unlikely(status < 0)) {
+            goto exit;
```

Comment:
1.  I’‘d use `return data;` directly instead of the indirection.
2.  Should an error message be logged somewhere?

Reply:
1. data is returned directly.
2. Added error message in error case